rio_tiler/io/base.py

```diff
@@ -39,21 +38,16 @@ class SpatialMixin:
 
     """
 
-    tms: TileMatrixSet = attr.ib(default=WEB_MERCATOR_TMS)
-
     bounds: BBox = attr.ib(init=False)
     crs: CRS = attr.ib(init=False)
 
     transform: Optional[Affine] = attr.ib(default=None, init=False)
     height: Optional[int] = attr.ib(default=None, init=False)
     width: Optional[int] = attr.ib(default=None, init=False)
 
-    geographic_crs: CRS = attr.ib(init=False, default=WGS84_CRS)
-
-    @cached_property
-    def geographic_bounds(self) -> BBox:
+    def geographic_bounds(self, crs: CRS = WGS84_CRS) -> BBox:
```

Comment:
I think the default CRS here should not be WGS84 but the CRS returned from `self.crs.geodetic_crs` As a asset _propable should_ have a CRS, the geographic crs here should be appropriate to the CRS of the dataset itself.

Reply:
@AndrewAnnex the problem is that rasterio CRS class doesn't have `geodetic_crs` 😬

Reply:
ahh well that would be a problem. But I have an idea and will post in the new issue